src/main/java/org/spongepowered/api/entity/projectile/DamagingProjectile.java

```diff
@@ -35,15 +38,57 @@
      * Gets the damage this projectile will deal to a {@link org.spongepowered.api.entity.living.Living}
      * if hit.
      *
-     * @return The damage of this arrow
+     * @return The damage to deal
      */
     double getDamage();
 
     /**
      * Sets the damage this projectile will deal to a LivingEntity if hit.
      *
-     * @param damage The damage
+     * @param damage The damage to deal
      */
     void setDamage(double damage);
 
+    /**
+     * Gets the damage this projectile will deal to the specified {@link EntityType}
+     * if hit.
+     *
+     * @param entityType The {@link EntityType} to set the damage amount for
+     * @return The damage to deal to the specified {@link EntityType}
+     */
+    double getDamageForEntityType(EntityType entityType);
+
+    /**
+     * Sets the damage this projectile will deal to the specified {@link EntityType}
+     * if hit.
+     *
+     * @param entityType The {@link EntityType} to set the damage amount for
+     * @param damage The damage to deal to the specified {@link EntityType}
+     */
+    void setDamageForEntityType(EntityType entityType, double damage);
+
+    /**
+     * Gets the damage this projectile will deal to the specified {@link EntityType}
+     * if hit.
+     *
+     * <p>The returned value takes precedence over
+     * {@link #getDamageForEntityType(EntityType)}.</p>
+     *
+     * @param entity The {@link Entity} to set the damage amount for
+     * @return The damage to deal to the specified {@link Entity}
+     */
+    double getDamageForEntity(Entity entity);
+
+    /**
+     * Gets the damage this projectile will deal to the specified {@link EntityType}
+     * if hit.
+     *
+     * <p>The specified damage value takes precedence over
+     * {@link #setDamageForEntityType(EntityType, double)}.</p>
+     *
+     * @param entity The {@link Entity} to set the damage amount for
+     * @param damage The damage to deal to the specified {@link Entity}
+     */
+    void setDamageForEntity(Entity entity, double damage);
```

Comment:
This could probably be done by just using an `EntityCollisionWithEntityEvent` event. I don't think there should be a property of a projectile whether it will damage a specific entity. If the entity dies, the projectile will still have a reference to that entity (OK it could be just a UUID but still wasted memory) but it will never exist again.

A plugin can easily store the UUID of the entity along with the damage to apply, and check when the collision event is fired.